packages/esm-patient-outcomes-app/src/smsform/send-sms-form.component.tsx

```diff
@@ -58,17 +69,23 @@ const SendSmsForm: React.FC<SendSmsFormProps> = ({
     promptBeforeClosing(() => isDirty);
   }, [isDirty, promptBeforeClosing, patient]);
 
+  useEffect(() => {
+    if (locale) {
+      methods.setValue('locale', locale);
+    }
+  }, [locale]);
```

Comment:
Instead of doing this, it's probably better to switch the `Select` to a controlled input (one where we set the value) [like this](https://github.com/openmrs/openmrs-esm-core/blob/533e15df65e4dcf0ef7cc4e7225b24353076fd62/packages/apps/esm-primary-navigation-app/src/components/change-language/change-language.modal.tsx#L26). This still results in default selecting the user's locale, but won't switch locales on this form if the user changes locales.